test/e2e/e2e.go

```diff
@@ -0,0 +1,204 @@
+	ctx := context.TODO()
+	req := testcontainers.ContainerRequest{
+		Image:        "livepeer/geth-with-livepeer-protocol:streamflow",
+		ExposedPorts: []string{"8546/tcp", "8545/tcp"},
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.NoError(t, err)
+
+	ip, err := container.Host(ctx)
+	require.NoError(t, err)
+
+	mappedPort, err := container.MappedPort(ctx, "8545")
+	require.NoError(t, err)
+
+	mappedPortWebServer, err := container.MappedPort(ctx, "8546")
+	require.NoError(t, err)
+
+	uri := fmt.Sprintf("http://%s:%s", ip, mappedPort.Port())
+	webServerUri := fmt.Sprintf("http://%s:%s", ip, mappedPortWebServer.Port())
+
+	return &gethContainer{Container: container, URI: uri, webServerURI: webServerUri}
+}
+
+func terminateGeth(t *testing.T, geth *gethContainer) {
+	err := geth.Terminate(context.TODO())
+	require.NoError(t, err)
+}
+
+// Start Livepeer helpers
+var (
+	httpPort = 8935
+	cliPort  = 7935
+	rtmpPort = 1935
+	mu       sync.Mutex
+)
+
+type livepeer struct {
+	dev   *devtool.Devtool
+	cfg   *starter.LivepeerConfig
+	ready chan struct{}
+}
+
+func lpCfg() starter.LivepeerConfig {
+	mu.Lock()
+	serviceAddr := fmt.Sprintf("127.0.0.1:%d", httpPort)
+	httpPort++
+	cliAddr := fmt.Sprintf("127.0.0.1:%d", cliPort)
+	cliPort++
+	rtmpAddr := fmt.Sprintf("127.0.0.1:%d", rtmpPort)
+	rtmpPort++
+	mu.Unlock()
+
+	ethPassword := ""
+	network := "devnet"
+	blockPollingInterval := 1
+	pricePerUnit := 1
+	initializeRound := true
+
+	cfg := starter.DefaultLivepeerConfig()
+	cfg.ServiceAddr = &serviceAddr
+	cfg.HttpAddr = &serviceAddr
+	cfg.CliAddr = &cliAddr
+	cfg.RtmpAddr = &rtmpAddr
+	cfg.EthPassword = &ethPassword
+	cfg.Network = &network
+	cfg.BlockPollingInterval = &blockPollingInterval
+	cfg.PricePerUnit = &pricePerUnit
+	cfg.InitializeRound = &initializeRound
+	return cfg
+}
+
+func startLivepeer(t *testing.T, lpCfg starter.LivepeerConfig, geth *gethContainer) *livepeer {
+	datadir := t.TempDir()
+	keystoreDir := filepath.Join(datadir, "keystore")
+	acc := devtool.CreateKey(keystoreDir)
+	devCfg := devtool.NewDevtoolConfig()
+	devCfg.Endpoint = geth.URI
+	devCfg.Account = acc
+	devCfg.KeystoreDir = keystoreDir
+
+	dev, err := devtool.Init(devCfg)
+	require.NoError(t, err)
+
+	err = dev.RequestTokens()
+	require.NoError(t, err)
+
+	err = dev.InitializeRound()
+	require.NoError(t, err)
+
+	lpCfg.EthUrl = &geth.URI
+	lpCfg.Datadir = &datadir
+	lpCfg.EthController = &dev.EthController
+	lpCfg.EthAcctAddr = &devCfg.Account
+
+	go func() {
+		starter.StartLivepeer(context.TODO(), lpCfg)
+	}()
+
+	ready := make(chan struct{})
+	go func() {
+		statusEndpoint := fmt.Sprintf("http://%s/status", *lpCfg.CliAddr)
+		var statusCode int
+		for statusCode != 200 {
+			time.Sleep(200 * time.Millisecond)
+			resp, err := http.Get(statusEndpoint)
+			if err == nil {
+				statusCode = resp.StatusCode
+			}
+		}
+		ready <- struct{}{}
+	}()
```

Comment:
Awesome! I was wondering how to get a signal when node is ready.

Reply:
Yeah, also for the context, this is probably good enough for E2E Tests, but we plan to add better health checks to Livepeer: 
- #2076
- #1920